### PR TITLE
wallet-ext: requires unlocked account for pages

### PR DESCRIPTION
**Pull request by pchrysochoidis**
## Description 

* disable access to pages that require unlocked accounts
* redirect to tokens page when account is locked

closes [APPS-1703](https://mysten.atlassian.net/browse/APPS-1703)

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/wallet/src/ui/app/hooks/useUnlockedGuard.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { useActiveAccount } from '../hooks/useActiveAccount';
+
+export function useUnlockedGuard() {
+	const navigate = useNavigate();
+	const { data: allAccounts, isLoading: isAccountsLoading } = useAccounts();
+	const activeAccount = useActiveAccount();
+	const loading = isAccountsLoading || !activeAccount;
+	const isInitialized = !!allAccounts?.length;
+	const isLocked = !!activeAccount?.isLocked;
+	const guardAct = !loading && isInitialized && isLocked;
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/hooks/useUnlockedGuard.ts:17]: could we rename `guardAct` to something a bit more clear like `guardAccount` or `isLocked` or `showGuard`

File: apps/wallet/src/ui/app/pages/home/apps/index.tsx [hunk 2]
```diff
@@ -66,6 +67,10 @@ function AppsPage() {
 		await setToSessionStorage<string>(APPS_PAGE_NAVIGATION, tag.link);
 	};
 
+	if (useUnlockedGuard()) {
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/pages/home/apps/index.tsx:70]: won't this hook redirect to the `/tokens` screen? do we need to `return null` everywhere if we're going to redirect when the account is locked?

**Review by mamos-mysten — APPROVED** [review]

File: apps/wallet/src/ui/app/hooks/useUnlockedGuard.ts (hunk 1 above)
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/hooks/useUnlockedGuard.ts:17]: The guard will act only when this variable is true. That's why I named it `guardAct`. Not sure what's a better name

File: apps/wallet/src/ui/app/pages/home/apps/index.tsx (hunk 2 above)
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/pages/home/apps/index.tsx:70]: It has a loading state, mainly until accounts are loaded. If that takes long returning null will avoid showing any contents of the actual page

**Pull request merged**